specification/monitor/resource-manager/Microsoft.Insights/preview/2023-06-01-preview/privateLinkScopes_API.json

```diff
@@ -0,0 +1,1166 @@
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "PrivateLinkScopedResources"
+        ],
+        "description": "Deletes an Azure monitor scoped resource with a given name.",
+        "operationId": "PrivateLinkScopedResources_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkScopeName"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "The name of the scoped resource object.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted scoped resource."
+          },
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Scoped resource does not exist."
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Deletes a scoped resource with a given name.": {
+            "$ref": "./examples/PrivateLinkScopedResourceDelete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/privateLinkScopes/{scopeName}/scopedResources": {
+      "get": {
+        "tags": [
+          "PrivateLinkScopedResources"
+        ],
+        "description": "Gets all scoped resources on a private link scope.",
+        "operationId": "PrivateLinkScopedResources_ListByPrivateLinkScope",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkScopeName"
+          },
+          {
+            "name": "kind",
```

Comment:
>kind [](http://example.com/codeflow?start=21&length=4)

use odata filtering instead

Reply:
I'm sorry, I'm not sure what you're talking about. What is odata filtering, and how does that differ from the current implementation?

Reply:
Please ref this doc: https://www.odata.org/getting-started/basic-tutorial/#queryData

Reply:
Based on the doc, I assume that the comment was meant to encourage the use of odata $filter rather than replacing $filter. At the same time, I believe that this kind of system is outside the scope of this PR. We are selecting between two different options, not filtering based on a detailed query with multiple different parameters. The goal of this new endpoint is to highlight the differences between the new and the old kind, and adding complex query syntax would be less intuitive to customers.

Reply:
Not using odata filtering to start with may result in making the query parameters complex as more requirements get added in future. Your requirements currently may be simple right now, but over time these may change.

Reply:
Since you are still in preview its the perfect time for you to switch to the odata standard to set your APIs for success for the future

Reply:
@rkmanda I don't believe this change is appropriate. Introducing OData filtering in the API will change the design and implementation of our service, which is not what we intended. We do not foresee this growing more complex, and we don't foresee the need for more flexible filtering options. This is just a toggle that was added to make clear to the end user that a new kind of scoped resource is introduced, and it's designed to be simple.

If you would like to discuss this further over call, I'll be happy to schedule some time to talk about it.

Reply:
The feedback we are giving is based on observation of many such patterns we have encountered in the past where a simple design grows into more complex ones and soon leads to bad customer experience. I dont think there is anything more to discuss. This is a strong recommendation but not a blocker so you may ship this